cmd/pileup/pileup.go

```diff
@@ -87,9 +90,17 @@ func pileup(infile string, outfile string, settings Settings) {
 func fmtOutput(pile sam.Pile, header sam.Header, builder *strings.Builder) string {
 	builder.Reset()
 	builder.WriteString(fmt.Sprintf("%s\t%d\t%d\t%d\t%d\t%d\t%d\t%d", header.Chroms[pile.RefIdx].Name, pile.Pos,
-		pile.Count[dna.A], pile.Count[dna.C], pile.Count[dna.G], pile.Count[dna.T], pile.Count[dna.N], pile.Count[dna.Gap]))
-	for seq, count := range pile.InsCount {
-		builder.WriteString(fmt.Sprintf("\t%s:%d", seq, count))
+		pile.CountF[dna.A]+pile.CountR[dna.A], pile.CountF[dna.C]+pile.CountR[dna.C], pile.CountF[dna.G]+pile.CountR[dna.G],
+		pile.CountF[dna.T]+pile.CountR[dna.T], pile.CountF[dna.N]+pile.CountR[dna.N], pile.CountF[dna.Gap]+pile.CountR[dna.Gap]))
+	for seq := range pile.InsCountF {
+		builder.WriteString(fmt.Sprintf("\t%s:%d", seq, pile.InsCountF[seq]+pile.InsCountR[seq]))
```

Comment:
Is the trick here that sometimes pile.InsCountR[seq] will miss in the map, but that during those misses we get 0 as the default value?

Reply:
Yes, that's right. That is why there is a loop on line 99 to get anything we missed.